### PR TITLE
make mobile extractors cover all possible layouts

### LINKED ISSUE #19: Mobile results are not working
Seems Google changed a layout of the pages...

### PR DESCRIPTION
**Pull request by pocesar**
determine the 3 mobile layouts available

* mobile page with javascript, "desktop-like"
* some google pages, like com.fj, displays as "googleweblight"
* most of the pages display as "mobile", with a responsive meta
* urls must be recreated com the `?q=` query parameter
* "next page" selector differed for each layout

right now the minimum is being returned:
https://pastebin.com/JjugQ45X

when everything is in place, I'll put it on the platform for testing then this can be merged

closes #19

### CODE REVIEW

**Comment by jancurn** [comment]
Great job @pocesar, is this still work in progress? Please let us know once it's ready for review. BTW it would be awesome if there were some unit tests, e.g. a directory where one could simply copy HTML files from Google Search result pages, and the test would just run the extractors on them so that you can check its output.

**Comment by pocesar** [comment]
@jancurn yes, it's still WIP. I can provide some HTML fixtures, do you mind I'm adding `jest` to the `devDependencies`?

**Comment by jancurn** [comment]
re jest - sure thing!

**Comment by pocesar** [comment]
https://my.apify.com/view/runs/dYpcrDmewpEFpHKau

tests are failing locally for the "mobile" type. cheerio is making Node.js crash with "heap out of memory"

**Comment by pocesar** [comment]
@jancurn guess it's good enough for now, if you want me to do more tests, let me know

mobile: https://my.apify.com/view/runs/aB9Aw2tDQDFFbPQZ3
desktop: https://my.apify.com/view/runs/NvfGS33ExgjNFPjHh

**Comment by pocesar** [comment]
`maxPagesPerQuery` is losing context on mobile, returning desktop results, I'm going to recreate the next page url from the INPUT parameters

**Comment by pocesar** [comment]
done, it's properly using the pagination now: 

mobile: https://my.apify.com/view/runs/f57mCwaGdRkTaDDaw
desktop: https://my.apify.com/view/runs/j3yWCAZkancbeiwYB

File: src/main.js [hunk 1]
```diff
@@ -34,6 +34,9 @@ Apify.main(async () => {
         requestList,
         requestQueue,
         maxConcurrency,
+        requestOptions: {
+            useMobileVersion: mobileResults,
```
Comment (jancurn) [inline src/main.js]: Is this needed? The request's HTTP headers are not used by Apify Proxy's Google SERP mode in any way.

Comment (pocesar) [reply src/main.js]: it isn't actually, after I saw the explanation on how the SERP proxy work, it ignores the user-agent headers. I'm removing this

**Pull request merged**